sympy/core/function.py

```diff
@@ -1005,10 +1005,15 @@ def __new__(cls, expr, *variables, **assumptions):
             variables = expr.free_symbols
             if len(variables) != 1:
                 from sympy.utilities.misc import filldedent
-                raise ValueError(filldedent('''
-                    Since there is more than one variable in the
-                    expression, the variable(s) of differentiation
-                    must be supplied to differentiate %s''' % expr))
+                if len(variables) == 0:
```

Comment:
@leosartaj the argument to diff is anything with _diff_wrt set to True. Presently diff doesn't handle such items correctly if they aren't set as variables in free_variables.